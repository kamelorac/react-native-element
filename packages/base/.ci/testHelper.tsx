import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
