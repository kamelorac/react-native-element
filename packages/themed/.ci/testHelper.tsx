import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
