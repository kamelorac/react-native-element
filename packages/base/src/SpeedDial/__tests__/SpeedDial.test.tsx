import React from 'react';
import { SpeedDial } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Speed Dial Component', () => {
  it('should match snapshot', () => {
    const { toJSON } = renderWithWrapper(
      <SpeedDial
        isOpen={true}
        icon={{ name: 'edit', color: '#fff' }}
        openIcon={{ name: 'close', color: '#fff' }}
      >
        <SpeedDial.Action
          icon={{ name: 'add', color: '#fff' }}
          title="Add"
          onPress={() => 'Add Something'}
        />
        <SpeedDial.Action
          icon={{ name: 'delete', color: '#fff' }}
          title="Delete"
          onPress={() => 'Delete Something'}
        />
      </SpeedDial>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
