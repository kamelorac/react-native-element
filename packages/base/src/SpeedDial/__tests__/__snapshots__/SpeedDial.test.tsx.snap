// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speed Dial Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    pointerEvents="box-none"
    style={
      [
        {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(36, 36, 36, 0.6)",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <RCTSafeAreaView
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        style={
          {
            "opacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "color": "black",
                  "elevation": 2,
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                undefined,
              ]
            }
          >
            Add
          </Text>
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "borderRadius": 28,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 2,
                  },
                  {
                    "borderRadius": 28,
                    "elevation": 4,
                  },
                  false,
                ]
              }
              testID="RNE_BUTTON_WRAPPER"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#ad1457",
                      "borderColor": "#2089dc",
                      "borderRadius": 2,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "padding": 8,
                      "paddingHorizontal": 10,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "height": 40,
                        "marginHorizontal": 5,
                        "overflow": "hidden",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                    testID="RNE__ICON__CONTAINER"
                  >
                    <View
                      accessibilityRole="button"
                      android_ripple={
                        {
                          "borderless": false,
                          "color": "rgba(0, 0, 0, 0.3)",
                          "radius": -5,
                        }
                      }
                      disabled={false}
                      testID="RNE__ICON__CONTAINER_ACTION"
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        testID="RNE__ICON"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            [
                              {
                                "color": "#fff",
                                "fontSize": 24,
                              },
                              {
                                "backgroundColor": "transparent",
                              },
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="RNE__ICON__Component"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        pointerEvents="auto"
        style={
          {
            "opacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 16,
                "marginRight": 24,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "color": "black",
                  "elevation": 2,
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                undefined,
              ]
            }
          >
            Delete
          </Text>
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "borderRadius": 28,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 2,
                  },
                  {
                    "borderRadius": 28,
                    "elevation": 4,
                  },
                  false,
                ]
              }
              testID="RNE_BUTTON_WRAPPER"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#ad1457",
                      "borderColor": "#2089dc",
                      "borderRadius": 2,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "padding": 8,
                      "paddingHorizontal": 10,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "height": 40,
                        "marginHorizontal": 5,
                        "overflow": "hidden",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                    testID="RNE__ICON__CONTAINER"
                  >
                    <View
                      accessibilityRole="button"
                      android_ripple={
                        {
                          "borderless": false,
                          "color": "rgba(0, 0, 0, 0.3)",
                          "radius": -5,
                        }
                      }
                      disabled={false}
                      testID="RNE__ICON__CONTAINER_ACTION"
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        testID="RNE__ICON"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            [
                              {
                                "color": "#fff",
                                "fontSize": 24,
                              },
                              {
                                "backgroundColor": "transparent",
                              },
                              {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              {},
                            ]
                          }
                          testID="RNE__ICON__Component"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 28,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 16,
            "marginTop": 0,
            "opacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 2,
              },
              {
                "borderRadius": 28,
                "elevation": 4,
              },
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#ad1457",
                  "borderColor": "#2089dc",
                  "borderRadius": 2,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 56,
                  "justifyContent": "center",
                  "padding": 16,
                  "paddingHorizontal": 10,
                  "width": 56,
                }
              }
            >
              <View
                style={
                  {
                    "height": 56,
                    "marginHorizontal": 5,
                    "overflow": "hidden",
                    "padding": 16,
                    "width": 56,
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        [
                          {
                            "color": "#fff",
                            "fontSize": 24,
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                          {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                      testID="RNE__ICON__Component"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
